src/Poseidon/CLI/List.hs

```diff
@@ -42,53 +42,104 @@ runList (ListOptions repoLocation listEntity rawOutput ignoreGeno) = do
-                pac <- allPackages
-                jannoRow <- posPacJanno pac
-                return $ IndividualInfo (posSamIndividualID jannoRow) (head (posSamGroupName jannoRow)) (posPacTitle pac)
+            return [(posPacTitle pac, posPacJanno pac) | pac <- allPackages]
     -- construct output
     hPutStrLn stderr "Preparing output table"
     (tableH, tableB) <- case listEntity of
         ListPackages -> do
             let tableH = ["Title", "Nr Individuals"]
-                tableB = do
-                    onePac <- groupBy (\x y -> indInfoPacName x == indInfoPacName y) $ 
-                        sortOn indInfoPacName allSampleInfo
-                    let pacTitle = indInfoPacName $ head onePac
-                        pacNrInds = show (length onePac)
-                    return [pacTitle, pacNrInds]
+                tableB = [[name, show (length rows)] | (name, rows) <- sortOn fst allSampleInfo]
             hPutStrLn stderr ("found " ++ show (length tableB) ++ " packages")
             return (tableH, tableB)
         ListGroups -> do
             let tableH = ["Group", "Packages", "Nr Individuals"]
+                pacJannoPairs = do
+                    (pacName, rows) <- allSampleInfo
+                    row <- rows
+                    return (pacName, row)
                 tableB = do
-                    oneGroup <- groupBy (\x y -> indInfoGroup x == indInfoGroup y) $ 
-                        sortOn indInfoGroup allSampleInfo
-                    let groupName = indInfoGroup $ head oneGroup
-                        groupPacs = intercalate "," $ nub $ map indInfoPacName oneGroup
+                    oneGroup <- groupBy (\x y -> (head . jGroupName . snd) x == (head . jGroupName . snd) y) $ 
+                        sortOn (head . jGroupName . snd) pacJannoPairs
+                    let groupName = head . jGroupName . snd . head $ oneGroup
+                        groupPacs = intercalate "," $ nub $ map fst oneGroup
                         groupNrInds = show (length oneGroup)
                     return [groupName, groupPacs, groupNrInds]
             hPutStrLn stderr ("found " ++ show (length tableB) ++ " groups/populations")
             return (tableH, tableB)
-        ListIndividuals -> do
-            let tableH = ["Package", "Individual", "Group"]
-            let tableB = do
-                    oneSample <- allSampleInfo
-                    return [indInfoPacName oneSample, indInfoName oneSample, indInfoGroup oneSample]
+        ListIndividuals moreJannoColumns -> do
+            let tableH = ["Package", "Individual", "Group"] ++ moreJannoColumns
+            tableB <- fmap concat . forM allSampleInfo $ \(pacName, rows) ->
+                forM rows (\row -> do
+                    moreFields <- extractAdditionalFields row moreJannoColumns
+                    return ([pacName, jIndividualID row, head (jGroupName row)] ++ moreFields))
```

Comment:
Displaying only the first GroupName makes sense, but it may hide the other names in this column. Of course one could get the full column with `-j Group_Name`, but I wonder if this should not just be the default with `--individuals`. 

For `--groups` we have to limit this to the first group name, I guess. But maybe the help text should then mention this.

Reply:
Yeah, I had thought about that too. Ideally we'd drop this behaviour of singling out the first group. In `list --packages`, we can just list all groups, in `list --groups` we would just list all the groups, even if some individuals appear in multiple groups`, and in `--individuals` there is no problem. For `list --groups` this would involve a slightly more involved fold. I'll submit a separate issue for that, as I think it's independent from this PR.